src/main/java/ch/psc/gui/components/signUp/CreateAccount.java

```diff
@@ -38,69 +49,143 @@ public CreateAccount() {
      * Initializes form and creates input field for creating a new account.
      */
     private void initialize() {
-        this.setSpacing(15);
+        this.setSpacing(40); //TODO: make appearance of textfields unified (in login and signup window)
         this.setPadding(new Insets(10, 20, 10, 20));
 
         usernameField.setPromptText(Config.getResourceText("signup.prompt.username"));
+        //Username field should not be empty
+        RequiredFieldValidator requiredUserNameValidator = new RequiredFieldValidator();
+        requiredUserNameValidator.setMessage(Config.getResourceText("signup.errorLabel.usernameRequired"));
+        usernameField.getValidators().add(requiredUserNameValidator);
+        usernameField.focusedProperty().addListener(new ChangeListener<Boolean>() {
+            @Override
+            public void changed(ObservableValue<? extends Boolean> observable, Boolean oldValue, Boolean newValue) {
+                if (!newValue) usernameField.validate(); //wenn kein neuer Value, dann validate
+            }
+        });
+
+
         emailTextField.setPromptText(Config.getResourceText("signup.prompt.email"));
+        //Email address field should not be empty
+        RequiredFieldValidator requiredEmailValidator = new RequiredFieldValidator();
+        requiredEmailValidator.setMessage(Config.getResourceText("signup.errorLabel.emailRequired"));
+        emailTextField.getValidators().add(requiredEmailValidator);
+        //Email address should be valid
+        RegexValidator inputMailValidator = new RegexValidator();
+        inputMailValidator.setMessage(Config.getResourceText("signup.errorLabel.emailNotValid"));
+        inputMailValidator.setRegexPattern("^[a-zA-Z0-9_!#$%&'*+/=?`{|}~^.-]+@[a-zA-Z0-9.-]+$");
```

Comment:
Eine eigene Email-Validator-Klasse. Die gleiche Prüfung wird an 2 Orten aufgebaut.
Zusammen in eine Klasse machen und dort dann auch das Pattern definieren.